lib/overcommit/hook_context/pre_push.rb

```diff
@@ -0,0 +1,32 @@
+module Overcommit::HookContext
+  # Contains helpers related to contextual information used by pre-push hooks.
+  class PrePush < Base
+    attr_accessor :args
+
+    def remote_name
+      @args[0]
+    end
+
+    def remote_url
+      @args[1]
+    end
+
+    def pushed_commits
+      input_lines.map do |line|
+        PushedCommit.new(*line.split(' '))
+      end
+    end
+
+    private
+
+    def input_lines
+      @input_lines ||= ARGF.read.split("\n")
```

Comment:
That's how I initially implemented it, but I didn't want this PR to grow beyond its intended scope. But I agree that constructor injection is the way to go here.